Penteract/Assets/Scripts/CameraController.cpp

```diff
@@ -42,12 +43,30 @@ void CameraController::Update() {
 	if (playerController == nullptr || transform == nullptr) return;
 	float3 playerGlobalPos = playerController->playerFang.playerMainTransform->GetGlobalPosition();
 
-
 	float3 desiredPosition = playerGlobalPos + float3(cameraOffsetX, cameraOffsetY, cameraOffsetZ);
-	float3 smoothedPosition = desiredPosition;
 
+	float3 smoothedPosition = desiredPosition;
+		
 	if (useSmoothCamera) {
-		smoothedPosition = float3::Lerp(transform->GetGlobalPosition(), desiredPosition, smoothCameraSpeed * Time::GetDeltaTime());
+		if (playerController->playerFang.isAiming()) {
+			float2 mousePosition = Input::GetMousePositionNormalized();
```

Comment:
Does this not take into account gamepad at all, is gamepad discarded?

Reply:
i dont have the xbox controller to addet yet, ill put this feature in TODO